### PR TITLE
ET-4403 Creating script for retrieving API calls

### PR DESCRIPTION
**Pull request by jorge-cr-13**
## What 🕵️ 🔍

- What does the pull request add and/or solve?

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

File: organization/biling/api_call.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    # Get the current year
+    current_year = datetime.now().year
+    
+    if month_number == 12:
+        current_year -= 1
+
+    # Create the start and end times
+    if month_number == 12:
+        current_year = current_year - 1
```
Comment (Robert-Steiner) [inline organization/biling/api_call.py:15]: That is one subtraction too much 
```suggestion
    # Create the start and end times
    if month_number == 12:
        current_year = current_year - 1
```

File: organization/billing/api_call.py
Comment (Robert-Steiner) [inline organization/billing/api_call.py:1]: two things:
- let's run a formatter over this file. We can use `black` (`pip install black`). run: `black .`
- if we select the current month, the number of requests from the beginning of the month to the current day should be returned.

**Comment by joecks** [comment]
What was the objective of this? In that form we have to run this script our selfs, right?
Wouldn't it make sense to deploy this as a lambda, that sends out an email at the end of the month? But I can agree to continue working on this and then eventually move this to modules and deploy this as a lambda. Wdyt?

**Comment by jorge-cr-13** [comment]
> What was the objective of this? In that form we have to run this script our selfs, right?
> Wouldn't it make sense to deploy this as a lambda, that sends out an email at the end of the month? But I can agree to continue working on this and then eventually move this to modules and deploy this as a lambda. Wdyt?

The main objective is to have the number of API calls per month so that at the beginning of the month, we can distribute the tenant's costs more precisely. But you are right the end goal is to have this automatized and sent to us on a monthly basis. But I think this is planned for later because also it is needed to be added the number of documents in postgres.

File: organization/billing/api_call.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        )
+        end_time = datetime(current_year, 1, 1, 0, 0).strftime("%Y-%m-%dT%H:%M:%SZ")
+    elif month_number == current_month:
+        start_time = datetime(current_year, month_number, 1, 0, 0).strftime(
+            "%Y-%m-%dT%H:%M:%SZ"
+        )
+        current_day = datetime.now().day
+        end_time = datetime(current_year, month_number, current_day - 1, 0, 0).strftime(
+            "%Y-%m-%dT%H:%M:%SZ"
+        )
```
Comment (Robert-Steiner) [inline organization/billing/api_call.py:26]:
```suggestion
        end_time =  datetime.now()
```

**Review by Robert-Steiner — COMMENTED** [review]
one small thing 
rest looks good

**Pull request merged**